integration_tests/src/main/python/data_gen.py

```diff
@@ -192,19 +194,24 @@ def with_special_pattern(self, pattern, flags=0, charset=sre_yield.CHARSET, weig
         instead of a hard coded string value.
         """
         strs = sre_yield.AllStrings(pattern, flags=flags, charset=charset, max_count=_MAX_CHOICES)
-        try:
-            length = int(len(strs))
-        except OverflowError:
-            length = _MAX_CHOICES
-        return self.with_special_case(lambda rand : strs[rand.randrange(0, length)], weight=weight)
+        length = strs.__len__()
+        def gen_str(rand):
+            index = rand.random() * length
+            return strs[int(index)]
+        return self.with_special_case(lambda rand : gen_str(rand), weight=weight)
 
     def start(self, rand):
-        strs = self.base_strs
-        try:
-            length = int(len(strs))
-        except OverflowError:
-            length = _MAX_CHOICES
-        self._start(rand, lambda : strs[rand.randrange(0, length)])
+        if self.pattern == "(.|\n){1,30}" and self.charset == sre_yield.CHARSET:
```

Comment:
Could we have a separate value for the default pattern so it is clear exactly what is happening?

Reply:
Done, use None as a default value now.